### PR TITLE
Support headers in service to service call/send

### PR DESCRIPTION
**Pull request by jackkleeman**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -31,6 +31,24 @@ pub struct Header {
     pub value: Cow<'static, str>,
 }
 
+impl From<service_protocol::messages::Header> for Header {
```
Comment (slinkydeveloper) [inline src/lib.rs:34]: Please move these two implementations under the comment "other conversions" in service_protocol/messages.rs

File: src/vm/mod.rs [hunk 2]
```diff
@@ -456,6 +456,11 @@ impl super::VM for CoreVM {
                 handler_name: target.handler,
                 key: target.key.unwrap_or_default(),
                 idempotency_key: target.idempotency_key,
+                headers: target
+                    .headers
+                    .into_iter()
+                    .map(crate::service_protocol::messages::Header::from)
```
Comment (slinkydeveloper) [inline src/vm/mod.rs:462]: I think you can do simply `Into::into`

**Review by slinkydeveloper — COMMENTED** [review]
LGTM, just small stuff and you can merge

File: src/vm/mod.rs (hunk 2 above)
Comment (jackkleeman) [reply src/vm/mod.rs:462]: i agree, but stylistically i much prefer seeing what type its becoming

Comment (tillrohrmann) [reply src/vm/mod.rs:462]: +1 for what Jack said. `Into::into` is really not helping for readability.

**Review by slinkydeveloper — APPROVED** [review]

**Pull request merged**